modules/gapi/include/opencv2/gapi/cpu/gcpukernel.hpp

```diff
@@ -269,31 +282,52 @@ template<typename U> struct get_out<cv::GOpaque<U>>
     }
 };
 
+template<typename, typename>
+struct OCVSetupHelper;
+
+template<typename Impl, typename... Ins>
+struct OCVSetupHelper<Impl, std::tuple<Ins...>>
+{
+    template<int... IIs>
+    static void setup_impl(GMetaArgs metaArgs, GArgs args, GArg &state, detail::Seq<IIs...>)
+    {
+        // unique_ptr <-> shared_ptr conversion ?
+        // To check: Conversion is possible only if the state which should be passed to
+        // 'setup' user callback isn't required to have previous value
+        std::shared_ptr<typename Impl::State> stPtr;
+        Impl::setup(detail::get_in_meta<Ins>(metaArgs, args, IIs)...,
+                    stPtr);
```

Comment:
This can remain one-liner. Your previous comment is longer than this call to `setup()`.

Reply:
Fixed.